### PR TITLE
Odin constant invalid read

### PR DESCRIPTION
**Pull request by jeanlego**
#### Motivation and Context
fix leak on arg_parse
double free guard to be used later

#### How Has This Been Tested?
odin pre_merge

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by jeanlego** [comment]
ping @ademmings

File: ODIN_II/SRC/ast_util.cpp [hunk 1]
```diff
@@ -1086,6 +1091,9 @@ static void check_binary_operation(ast_node_t **node)
  *-------------------------------------------------------------------------------------------*/
 ast_node_t *fold_unary(ast_node_t *node)
 {
+	if(!node)
```
Comment (ademmings) [inline ODIN_II/SRC/ast_util.cpp]: if you're gonna check this on fold_unary, you should check it on fold_binary too please

Comment (jeanlego) [reply ODIN_II/SRC/ast_util.cpp]: I've made both function more consistent with each other. last commit should resolve your comment

**Comment by jeanlego** [comment]
@ademmings should be good to go

**Pull request merged**